client/src/pages/LoginPage.js

```diff
@@ -0,0 +1,198 @@
+import React, { useEffect, useContext } from "react";
+import Grid from "@material-ui/core/Grid";
+import { Card, Snackbar, TextField } from "@material-ui/core";
+import { Button } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { Typography } from "@material-ui/core";
+import { Link } from "react-router-dom";
+import { useFormik } from "formik";
+import * as yup from "yup";
+import Alert from "../components/Alert";
+import { UserContext } from "../App"
+
+
+
+const validationSchema = yup.object({
+  email: yup
+    .string("Enter your email")
+    .email("Enter a valid email")
+    .required("Email is required"),
+  password: yup
+    .string("Enter your password")
+    .min(6, "Password should be of minimum 6 characters length")
+    .required("Password is required"),
+});
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    flexGrow: 1,
+    "margin-top": "5rem",
+    "& .MuiOutlinedInput-root": {
+      "& fieldset": {
+        borderRadius: 50,
+      },
+    },
+  },
+  padding: {
+    padding: "5rem",
+    "padding-top": "3rem",
+  },
+  roundedCorner: {
+    borderRadius: 50,
+  },
+  fullWidth: {
+    width: "100%",
+  },
+  marginBottom: {
+    "margin-bottom": "2rem",
+  },
+}));
+
+
+const timer = (t) => {
+  return new Promise((res) => {
+    setTimeout(() => res(), t);
+  });
+};
+
+export default function LoginPage() {
+  const { vertical, horizontal } = { vertical: "bottom", horizontal: "center" };
+
+  const {state,dispatch} = useContext(UserContext)
```

Comment:
Hey man! I wanted to suggest changing the `useContext` to `useReducer` instead. It looks like you're using it as a reducer. I don't know if @jessdesr agrees. Everything else would work the same way, but without the performance implications of `useContext` (which will re-render components that use the context every time the context changes...  even if they are memozied).

Here's what it would look like:

```javascript
import { reducer, initialState } from '../store/UserReducer';
// ...
const [state, dispatch] = useReducer(reducer, initialState);
```

And it' would be the same in the SignupPage.js.

Reply:
@johnsciutto is right, useReducer is the correct hook here